### ISSUE TITLE
cargo test doesn't check against conjure

### ISSUE THREAD

**Issue by YehorBoiar** [conjure-cp/conjure-oxide#508]
Conjure-oxide checks solutions against savile row only when ACCEPT=true is passed, but shouldn't it always do that?

I was thinking that because in PR #435 I currently have a couple of solutions that don't match conjure, but it shows that all tests are being passed. It could lead to some problems in future PRs. However, I'm not sure about it, what do you think? @niklasdewally

**Comment by niklasdewally** [comment]
I presume you mean Conjure? 

When you run ACCEPT=true, it should save the expected solutions to the file, checking these against Conjure. When testing our model we compare our solutions to these. As they have already been checked by Conjure during code-gen, we shouldn't need to do so again?

**Comment by niklasdewally** [comment]
The fact that the integration tests pass but conjure fails is definitely a bug though.

We have started having issues with CI tests failing for no reason, in particular macos-latest, which might be related

**Comment by ozgurakgun** [comment]
I agree we shouldn't run conjure every time we run integration tests since the output shouldn't change.

If there is a case where it did, that would definitely be a bug (or some sort of a major conjure version upgrade, but that didn't happen)

@YehorBoiar can you explain what's happening? How can we reproduce the issue?

**Comment by YehorBoiar** [comment]
Here is the bug:

When I run `ACCEPT=true cargo test` I get my `tests_integration_basic_max_02` failed because solutions don't match conjure
![image](https://github.com/user-attachments/assets/3aa729a5-05ec-4109-a834-32f0fd9404f8)

However, despite that, my expected `parse.serialised`, `rewrite.serialised`, and `minion.solutions` are still being rewritten to its genereted files. So next time I run `cargo test` it shows that my `tests_integration_basic_max_02` is fine.
![image](https://github.com/user-attachments/assets/7fde4a95-8d25-470a-9427-024986dfe258)

**Comment by ozgurakgun** [comment]
All I can say without looking too much into it is taht `a=3` shouldn't be a solution. the test case Essence is:

```
find a : int(0..3)
such that max([2,a]) <= 2
```

Something is going wrong, not sure what.

**Comment by YehorBoiar** [comment]
I know that solution isn't correct. However, I want to point out a separate issue.  We still rewrite from generated model into an expected model even though the test isn't being passed. I was expecting that when we pass an `ACCEPT=true` it would rewrite only those generated models that have passed the test. Is my expectation reasonable? Or `accept` flag should work this way?

**Comment by ozgurakgun** [comment]
wait a minute. conjure generating an incorrect solution would break our flow. why do we get a=3 as a solution from conjure?

can you check `conjure solve --number-of-solutions=all FILENAME` on that Essence file to see why this is happening?

**Comment by ozgurakgun** [comment]
I see, thanks for the investigation. It sounds like we should only accept the outputs if the tests pass, indeed. Do you want to work on a fix?

**Comment by YehorBoiar** [comment]
I can do that
